lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -798,6 +828,182 @@ mkShelleyWallet ctx wid cp meta pending progress = do
             , changesAt = mepochInfo
             }
 
+--------------------- Shared Wallet
+
+postSharedWallet
+    :: forall ctx s k n.
+        ( s ~ SharedState n k
+        , ctx ~ ApiLayer s k
+        , SoftDerivation k
+        , MkKeyFingerprint k (Proxy n, k 'AddressK XPub)
+        , MkKeyFingerprint k Address
+        , WalletKey k
+        , HasDBFactory s k ctx
+        , HasWorkerRegistry s k ctx
+        , Typeable s
+        , Typeable n
+        )
+    => ctx
+    -> ((SomeMnemonic, Maybe SomeMnemonic) -> Passphrase "encryption" -> k 'RootK XPrv)
+    -> (XPub -> k 'AccountK XPub)
+    -> ApiSharedWalletPostData
+    -> Handler ApiSharedWallet
+postSharedWallet ctx generateKey liftKey postData =
+    case postData of
+        ApiSharedWalletPostData (Left body) ->
+            postSharedWalletFromRootXPrv ctx generateKey body
+        ApiSharedWalletPostData (Right body) ->
+            postSharedWalletFromAccountXPub ctx liftKey body
+
+postSharedWalletFromRootXPrv
+    :: forall ctx s k n.
+        ( s ~ SharedState n k
+        , ctx ~ ApiLayer s k
+        , SoftDerivation k
+        , MkKeyFingerprint k (Proxy n, k 'AddressK XPub)
+        , MkKeyFingerprint k Address
+        , WalletKey k
+        , HasDBFactory s k ctx
+        , HasWorkerRegistry s k ctx
+        , Typeable s
+        , Typeable n
+        )
+    => ctx
+    -> ((SomeMnemonic, Maybe SomeMnemonic) -> Passphrase "encryption" -> k 'RootK XPrv)
+    -> ApiSharedWalletPostDataFromMnemonics
+    -> Handler ApiSharedWallet
+postSharedWalletFromRootXPrv ctx generateKey body = do
+    let state = mkSharedStateFromRootXPrv (rootXPrv, pwd) accIx g pTemplate dTemplateM
+    void $ liftHandler $ createWalletWorker @_ @s @k ctx wid
+        (\wrk -> W.createWallet  @(WorkerCtx ctx) @s @k wrk wid wName state)
+        (\wrk _ -> W.manageRewardBalance @(WorkerCtx ctx) @s @k (Proxy @n) wrk wid)
```

Comment:
Note: the implementation of that function, or more exactly, of `readRewardAccount` used by this function should likely be updated. Reward accounts for shared wallets are captured by the delegation part (which can be missing).

Reply:
used `idleWorker` to be consistent. to be treated in the next PR